lib/jwt/jwa/ecdsa.rb

```diff
@@ -2,8 +2,35 @@
 
 module JWT
   module JWA
-    module Ecdsa
-      module_function
+    class Ecdsa
+      include JWT::JWA::SigningAlgorithm
+
+      def initialize(alg, digest)
+        @alg = alg
+        @digest = OpenSSL::Digest.new(digest)
+      end
+
+      def sign(data:, signing_key:)
+        curve_definition = curve_by_name(signing_key.group.curve_name)
```

Comment:
Unrelated to the changes: There is not validation that the provided `signing_key` is of the expected type

Reply:
Lets improve ecdsa later